arelle/plugin/validate/EDINET/PluginValidationDataExtension.py

```diff
@@ -194,6 +204,12 @@ def getUploadFilepaths(self, modelXbrl: ModelXbrl) -> list[Path]:
             paths.update(path.parents)
         return sorted(paths)
 
+    def hasRequiredValidFact(self, modelXbrl: ModelXbrl, qname: QName) -> bool:
+        requiredFacts = modelXbrl.factsByQname.get(qname, set())
+        if not len([fact for fact in requiredFacts if fact is not None and fact.xValid >= VALID and not fact.isNil]):
+            return False
+        return True
```

Comment:
* It's more efficient to stop checking facts as soon as a match is found.
* The name of the function should mention that this is looking for valid non-nil facts, not just valid facts.
* factsByQname values should never be None.
```suggestion
    def hasValidNonNilFact(self, modelXbrl: ModelXbrl, qname: QName) -> bool:
        requiredFacts = modelXbrl.factsByQname.get(qname, set())
        return any(
            fact.xValid >= VALID and not fact.isNil
            for fact in requiredFacts
        )
```